File: utio/models/SensorData.js
/**
 * 传感器数据模型
 * 对应表：sensor_data
 */

class SensorData {
  constructor(data = {}) {
    this.id = data.id || null
    this.recordTime = data.record_time || data.recordTime || null
    this.temperatureC = data.temperature_c || data.temperatureC || 0
    this.soilMoisturePct = data.soil_moisture_pct || data.soilMoisturePct || 0
    this.lightLux = data.light_lux || data.lightLux || 0
    this.isRaining = data.is_raining !== undefined ? data.is_raining : (data.isRaining !== undefined ? data.isRaining : 0)
    this.createdAt = data.created_at || data.createdAt || null
    this.updatedAt = data.updated_at || data.updatedAt || null
  }

  /**
   * 转换为数据库字段格式
   */
  toDbFormat() {
    return {
      record_time: this.recordTime,
      temperature_c: this.temperatureC,
      soil_moisture_pct: this.soilMoisturePct,
      light_lux: this.lightLux,
      is_raining: this.isRaining ? 1 : 0,
    }
  }

  /**
   * 转换为前端格式
   */
  toFrontendFormat() {
    return {
      id: this.id,
      time: this.recordTime ? new Date(this.recordTime).toISOString() : null,
      temperatureC: parseFloat(this.temperatureC),
      soilMoisturePct: parseFloat(this.soilMoisturePct),
      lightLux: parseInt(this.lightLux),
      isRaining: this.isRaining ? 1 : 0,
      createdAt: this.createdAt,
      updatedAt: this.updatedAt,
    }
  }

  /**
   * 验证数据有效性
   */
  validate() {
    const errors = []
    if (!this.recordTime) errors.push('记录时间不能为空')
    if (this.temperatureC < -50 || this.temperatureC > 100) errors.push('温度值超出合理范围')
    if (this.soilMoisturePct < 0 || this.soilMoisturePct > 100) errors.push('土壤湿度值超出合理范围')
    if (this.lightLux < 0) errors.push('光照强度不能为负数')
    return errors
  }
}

module.exports = SensorData

File: backend/src/main/java/com/greenhouse/entity/SensorData.java
package com.greenhouse.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.util.Date;

/**
 * 传感器数据实体类
 * 对应表：sensor_data
 */
@Data
@TableName("sensor_data")
public class SensorData {
    /**
     * 主键ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 记录时间（精确到小时）
     */
    private Date recordTime;

    /**
     * 温度（摄氏度）
     */
    private BigDecimal temperatureC;

    /**
     * 湿度（百分比）
     */
    private BigDecimal humidityPct;

    /**
     * 土壤湿度（百分比）
     */
    private BigDecimal soilMoisturePct;

    /**
     * 光照强度（lux）
     */
    private Integer lightLux;

    /**
     * 是否下雨（0否/1是）
     */
    private Boolean isRaining;

    /**
     * 氧气含量（百分比）
     */
    private BigDecimal oxygenPct;

    /**
     * 二氧化碳含量（ppm）
     */
    private Integer co2Ppm;

    /**
     * 创建时间
     */
    private Date createdAt;

    /**
     * 更新时间
     */
    private Date updatedAt;
}

